### PR TITLE
Add hiding and showing CFU student responses

### PR DESCRIPTION
**Pull request by lfryemason**
Adds functionality to the 'hide response' button in the previously added dropdown
Adds an alert to show all hidden responses

https://github.com/user-attachments/assets/fc172adc-f50a-465b-9c65-3b373fb7d931

## Links
[TEACH-1130](https://codedotorg.atlassian.net/browse/TEACH-1130)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dju90** [comment]
Looks great! I'm assuming the purple color of the kebab menu will make more sense once the response boxes are also changed to lighter purple.

File: apps/src/templates/levelSummary/ResponseMenuDropdown.jsx [hunk 1]
```diff
@@ -43,7 +43,10 @@ const ResponseMenuDropdown = ({response}) => {
               <button
                 className={styles.dropdownOption}
                 type="button"
-                onClick={() => setIsOpen(false)}
+                onClick={() => {
+                  setIsOpen(false);
+                  hideResponse(response);
```
Comment (kobryan0619) [inline apps/src/templates/levelSummary/ResponseMenuDropdown.jsx]: I am wondering if we need to pass through the whole `response` here (both in the component and in this function) or just the `user_id` of the response to be hidden.

File: apps/test/unit/templates/levelSummary/FreeResponseResponsesTest.jsx [hunk 2]
```diff
@@ -1,26 +1,55 @@
-import {shallow} from 'enzyme'; // eslint-disable-line no-restricted-imports
+import {render, screen} from '@testing-library/react';
```
Comment (kobryan0619) [inline apps/test/unit/templates/levelSummary/FreeResponseResponsesTest.jsx:1]: Thanks for changing this over to React Testing Library

**Review by kobryan0619 — APPROVED** [review]
Approved - one clarifying question, but assuming there is a reason for that choice, it looks good to go.

**Review by levadadenys — APPROVED** [review]
LGTM from Design System side! Thanks for this update!

**Pull request merged**